gematria/datasets/process_and_filter_bbs_lib.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gematria/datasets/process_and_filter_bbs_lib.h"
+
+#include "X86InstrInfo.h"
+#include "gematria/llvm/disassembler.h"
+#include "gematria/llvm/llvm_architecture_support.h"
+#include "gematria/utils/string.h"
+#include "llvm/ADT/StringExtras.h"
+
+using namespace llvm;
+
+namespace gematria {
+
+BBProcessorFilter::BBProcessorFilter() {
+  LLVMSupport = LlvmArchitectureSupport::X86_64();
+  InstructionPrinter = LLVMSupport->CreateMCInstPrinter(0);
+}
+
+Expected<std::string> BBProcessorFilter::processBasicBlock(
```

Comment:
We should rename this to `removeRiskyInstructions()` so that the name describes more precisely what the function does.

Also, in the future we might also want to do other things here, e.g. break things down into multiple basic blocks rather than just remove the instructions and we might need multiple filtering functions with different, more descriptive, names.